src/menu/mod.rs

```diff
@@ -0,0 +1,78 @@
+use bevy::app::AppExit;
+use bevy::prelude::*;
+
+use crate::AppState;
+
+pub mod main_menu;
+pub mod settings;
+
+pub struct MenuAssets {
+    button_normal: Handle<ColorMaterial>,
+    button_hover: Handle<ColorMaterial>,
+    button_active: Handle<ColorMaterial>,
+
+    menu_panel_background: Handle<ColorMaterial>,
+
+    transparent: Handle<ColorMaterial>,
+
+    font_light: Handle<Font>,
+    font_light_italic: Handle<Font>,
+    font_regular: Handle<Font>,
+    font_regular_italic: Handle<Font>,
+    font_bold: Handle<Font>,
+    font_bold_italic: Handle<Font>,
+}
+
+impl FromResources for MenuAssets {
+    fn from_resources(resources: &Resources) -> Self {
+        let mut materials = resources.get_mut::<Assets<ColorMaterial>>().unwrap();
+        let mut assets = resources.get_mut::<AssetServer>().unwrap();
+
+        MenuAssets {
+            button_normal: materials.add(Color::rgb(0.3, 0.3, 0.36).into()),
+            button_hover: materials.add(Color::rgb(0.4, 0.4, 0.46).into()),
+            button_active: materials.add(Color::rgb(0.24, 0.24, 0.32).into()),
+
+            menu_panel_background: materials.add(Color::rgb(0.2, 0.2, 0.24).into()),
+
+            transparent: materials.add(Color::NONE.into()),
+
+            font_light: assets.load("fonts/sansation/sansation_light.ttf"),
+            font_light_italic: assets.load("fonts/sansation/sansation_light_italic.ttf"),
+            font_regular: assets.load("fonts/sansation/sansation_regular.ttf"),
+            font_regular_italic: assets.load("fonts/sansation/sansation_italic.ttf"),
+            font_bold: assets.load("fonts/sansation/sansation_bold.ttf"),
+            font_bold_italic: assets.load("fonts/sansation/sansation_bold_italic.ttf"),
+        }
+    }
+}
+
+pub enum ClickAction {
+    ChangeState(AppState),
+    Exit,
+}
+
+pub fn button_interact(
+    mut state: ResMut<State<AppState>>,
+    mut app_exit: ResMut<Events<AppExit>>,
+    materials: Res<MenuAssets>,
+    mut query: Query<
+        (&Interaction, &mut Handle<ColorMaterial>, &ClickAction),
+        (Mutated<Interaction>, With<Button>),
+    >,
+) {
+    for (interaction, mut material, action) in query.iter_mut() {
+        match interaction {
+            Interaction::Clicked => {
+                *material = materials.button_active.clone();
+
+                match action {
```

Comment:
Having one system handle all button presses isn't gonna scale. We would need to add more and more parameters to this function in the future as we add more buttons to the game.

However, I think you can leave this for now. I have a proposal for how to solve this elegantly, and I can submit that as a subsequent PR after we merge this one.

Reply:
Yup, this isn't intended to be a "final" solution in any way, I just wanted to make the menu buttons work 😄

Reply:
Yep, this is the mindset to have. None of what we are doing here is "final", just have fun and make it work :smile: 

But I also feel obligated to leave notes about important points that we should all be aware of, so that my thoughts are put down in writing.